cpp/tests/interop/arrow_utils.hpp

```diff
@@ -176,3 +176,28 @@ std::shared_ptr<arrow::Array> get_arrow_list_array(
 
 std::pair<std::unique_ptr<cudf::table>, std::shared_ptr<arrow::Table>> get_tables(
   cudf::size_type length = 10000);
+
+template <typename T>
+[[nodiscard]] auto make_decimal128_arrow_array(std::vector<T> const& data,
+                                               std::optional<std::vector<int>> const& nulls,
+                                               int32_t scale) -> std::shared_ptr<arrow::Array>
+{
+  auto constexpr SIZE_OF_INT128  = 16;
+  auto constexpr BIT_WIDTH_RATIO = SIZE_OF_INT128 / sizeof(T);
+
+  std::shared_ptr<arrow::Array> arr;
+  arrow::Decimal128Builder decimal_builder(arrow::decimal(18, -scale),
+                                           arrow::default_memory_pool());
+
+  for (T i = 0; i < static_cast<T>(data.size() / BIT_WIDTH_RATIO); ++i) {
+    if (nulls.has_value() and not nulls.value()[i]) {
+      decimal_builder.AppendNull();
```

Comment:
If `not nulls.value[i]` means it's a null then `nulls` should be renamed to `validity` or something.